pkg/storaged/resize.jsx

```diff
@@ -270,16 +317,31 @@
     });
 }
 
-export function shrink_dialog(client, lvol, info, to_fit) {
-    const block = client.lvols_block[lvol.path];
-    const vgroup = client.vgroups[lvol.VolumeGroup];
+export function shrink_dialog(client, lvol_or_part, info, to_fit) {
+    let title, block, name, orig_size, round_size;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        const vgroup = client.vgroups[lvol_or_part.VolumeGroup];
+
+        title = _("Shrink logical volume");
+        block = client.lvols_block[lvol_or_part.path];
+        name = lvol_or_part.Name;
+        orig_size = lvol_or_part.Size;
+        round_size = vgroup.ExtentSize;
+    } else {
+        title = _("Shrink partition");
+        block = client.blocks[lvol_or_part.path];
+        name = block_name(block);
+        orig_size = lvol_or_part.Size;
+        round_size = 1024 * 1024;
+    }
 
     const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
                                    _("shrink"));
 
     if (usage.Blocking) {
         dialog_open({
-            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Title: cockpit.format(_("$0 is in use"), name),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230928-092115-8ec73ebe-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)